plugins/env/app/models/environment_variable.rb

```diff
@@ -92,13 +94,13 @@ def resolve_secrets(project, deploy_group, env, preview:)
     end
   end
 
-  # used by `priority` from GroupScope
-  def project?
-    parent_type == "Project"
+  def priority
+    [PARENT_PRIORITY.index(parent_type) || 999, super]
```

Comment:
why use a lookup here when you used a case statement in another class?
also didn't you get rid of the array?

Reply:
the base class uses an integer and the sub-classes change it to an array so they can prepend their own priority

Reply:
hmmm yeah I'll change the other one to array too

Reply:
can you put that in a comment somewhere too...